iroh-rpc/src/stream.rs

```diff
@@ -292,3 +162,85 @@ pub struct StreamConfig {
     pub peer_id: PeerId,
     pub channel: mpsc::Sender<Command>,
 }
+
```

Comment:
thought it might be best to pull all stream/associated things into here

Also, realized that I was actually making closing the stream more complicated than it needed to be since we know how many packets of data we are expecting, we can just manage when it should close.